repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -102,7 +105,30 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
         wrappedRequest.setRequestURI(joinTokensIntoPath(uriTokens))
       }
 
-      filterChain.doFilter(wrappedRequest, wrappedResponse)
+      val requestLinkPaths = config.getLinkResource
+        .filter(resource => resource.getUriPathRegex =~ wrappedRequest.getRequestURI)
+        .filter(resource => isMatchingMethod(resource.getHttpMethods.toSet, wrappedRequest.getMethod))
+        .map(resource => getPathsForContentType(wrappedRequest.getContentType, resource.getRequest))
+        .fold(List.empty[LinkPath])(_ ++ _)
```

Comment:
As for why, I have no idea.

Maybe because `flatMap` returns a `Buffer` in this case and my brain got confused? Or because I was just really into `fold` that day?